spec/services/task_services_spec.rb

```diff
@@ -0,0 +1,82 @@
+require 'rails_helper'
+
+RSpec.describe TaskService do
+  let(:service) { TaskService.new }
+  let(:user) { FactoryBot.create(:user) }
+  let(:admin) { FactoryBot.create(:user, role: 0, email: 'admin@gmail.com') }
+  let!(:task_1) { FactoryBot.create(:task, title: 'task1', assignee_user_id: user.id, assigner_user_id: admin.id) }
+  let!(:task_2) { FactoryBot.create(:task, title: 'task2', assignee_user_id: user.id, assigner_user_id: admin.id) }
+  let(:category) { FactoryBot.create(:task_category) }
+
+  context 'for creating a task' do
+    it 'creates a task with valid attributes' do
+      task_params = { title: 'task_demo', description: 'dd', assignee_user_id: user.id, due_date: "23/08/2024" }
+      expect { service.create(admin, task_params) }.to change(Task, :count).by(1)
+    end
+
+    it 'handles errors and sets errors attribute' do
+      task_params = { title: 'Test Task', description: 'dd', assignee_user_id: 12, due_date: "23/08/2024" } 
+      service.create(admin, task_params)
+      expect(service.errors).to be_an_instance_of(ActiveRecord::RecordInvalid)
```

Comment:
are we storing ruby exception objects in service errors? shouldn't those be strings?